src/Perlang.Tests.Integration/Operator/Binary/Addition.cs

```diff
@@ -1,44 +1,69 @@
+// Needed to be able to set CultureInfo.CurrentCulture without leaking the change to other test methods.
+
 #pragma warning disable CS1998
 
 using System.Globalization;
 using System.Threading.Tasks;
+using FluentAssertions;
 using Xunit;
 using static Perlang.Tests.Integration.EvalHelper;
 
-namespace Perlang.Tests.Integration.Operator
+namespace Perlang.Tests.Integration.Operator.Binary
 {
     public class Addition
     {
-        [Fact]
-        public void addition_of_strings_performs_concatenation()
+        [Theory]
+        [MemberData(nameof(BinaryOperatorData.Plus_PlusEqual_result), MemberType = typeof(BinaryOperatorData))]
+        void addition_of_numbers_performs_numeric_addition(string i, string j, string expectedResult)
         {
-            string source = @"
-                var s1 = ""foo"";
-                var s2 = ""bar"";
+            string source = $@"
+                var i1 = {i};
+                var i2 = {j};
 
-                print s1 + s2;
+                print i1 + i2;
             ";
 
             string result = EvalReturningOutputString(source);
-            Assert.Equal("foobar", result);
+
+            result.Should()
+                .Be(expectedResult);
+        }
+
+        [Theory]
+        [MemberData(nameof(BinaryOperatorData.Plus_PlusEqual_type), MemberType = typeof(BinaryOperatorData))]
+        void addition_of_numbers_returns_expected_type(string i, string j, string expectedType)
+        {
+            string source = $@"
+                var i1 = {i};
+                var i2 = {j};
+
+                print (i1 + i2).get_type();
+            ";
+
+            string result = EvalReturningOutputString(source);
+
+            result.Should()
+                .Be(expectedType);
         }
```

Comment:
Should we add a `unsupported_types_emits_expected_errors`-style test here also? I think it would make sense.

Reply:
Has been fixed, resolving.